### ISSUE TITLE
Fatal error:  Call to undefined function pcntl_exec() in ncludes/startup.inc on line 256

### ISSUE THREAD

**Issue by vijaycs85** [drush-ops/drush#1657]

**Comment by weitzman** [comment]
Yes, pcntl extension is now a requirement of drush, though you can work around it. Where did you get your PHP (homebrew, MAMP, docker, etc)?

**Comment by vijaycs85** [comment]
MAMP!

**Comment by vijaycs85** [comment]
Found https://github.com/drush-ops/drush/pull/1595#issuecomment-144810398 and this looks like a workaround:

```PHP
  if (!extension_loaded('pcntl')) {
    $command = $found_script . ' ' . implode(' ', $arguments);
    passthru(trim($command), $error);
    exit($error);
  }
```

**Comment by weitzman** [comment]
Or you try these to cleanly redispatch:

* http://adampatterson.ca/blog/2014/10/laravel-tinker-boris-mamp-os-x-yosemite/
* http://stackoverflow.com/questions/5230255/how-to-enable-process-control-extension-pcntl-in-php-mamp

**Comment by xjm** [comment]
@vijaycs85 Where would one add said workaround?

**Comment by vijaycs85** [comment]
@xjm  https://github.com/drush-ops/drush/pull/1659

**Comment by vijaycs85** [comment]
@weitzman thanks for the links. I followed http://stackoverflow.com/questions/5230255/how-to-enable-process-control-extension-pcntl-in-php-mamp and now I get neither fatal nor any output. I will keep looking and post, if I have any update.

**Comment by xjm** [comment]
FWIW I tried the workaround above, as well as using drush.launch and drush.php directly, and while all three removed the fatal, I still cannot use drush si with --db-url but get "Drush 8.0-dev does not support Drupal . / Command site-install needs a higher bootstrap level to run - you will need to invoke drush from a more functional Drupal environment to run this command. The drush command 'si' could not be executed." I'm not sure if this is the same bug or a different one, so I haven't invested the time yet to go through installing pcntl.

**Comment by weitzman** [comment]
`drush.launcher st` is working for me.

**Comment by weitzman** [comment]
Actually, when I call `drush.launcher updb` I get a redispatch to `drush` so my suggestion needs work /cc @greg-1-anderson

**Comment by webchick** [comment]
I thought that I was ok here (my last drush si must've been a few days back, before D8's conf_path() removal, which prompted my upgrade) but actually DevDesktop / internal PHP (OSX Yosemite) doesn't seem to ship with this extension either.

Despite #1652, composer global update didn't do anything for me, until I manually edited /Users/webchick/.composer/composer.json to add the same "require" line. (That was probably not the right way to do it, but there is no mention of pcntl on the installation/upgrading instructions http://docs.drush.org/en/master/install/.)

Now I do get an actual error:

Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - The requested PHP extension ext-pcntl * is missing from your system.

But no idea how to remedy that.

In another issue I can't find there was talk of adding a fallback for Windows; seems like it might be a good idea to add a fallback for at least Mac as well.

**Comment by weitzman** [comment]
@webchick - my dev desktop has pcntl. i have: Built: Aug 13 2015 08:52:31

Any fallback we add would work for mac users as well. Subscibe to #1660

**Comment by greg-1-anderson** [comment]
Fixed by #1660.  Windows not covered, but this should take care of Mac & Linux users. For now, we are presuming that Windows users will be using drush.bat instead of the Drush finder.  The Drush finder might alwasy work on the Git Bash shell provided with Git for Windows; haven't tried it, though.

Please open a separate issue for Windows problems.

**Comment by webchick** [comment]
Working again; thanks a lot! I will also update my DevDesktop. :)
